src/components/game/coreState/utils/FillCheck.js

```diff
@@ -87,17 +93,28 @@ export const checkFilledLines = ({
       }
     }
   }
+  scorekeeper.scoreFilledLines(linesCleared);
 };
 
 // Check all filled targets, remove them from targetBlocks, and erase all
 // covered cells to replace with a call to emptyValue
-export const advanceAndCheckTargets = ({ targets, board, emptyValue }) => {
+export const advanceAndCheckTargets = ({ targets, board, emptyValue, coreState, scorekeeper }) => {
   var gameOver = false;
   targets.forEach((target) => target.advanceUpdate());
   // Clear the targets in a 2nd pass so that the player can hit combos on targets in the same move.
+  var clearedTargets = 0;
   targets.forEach((target) => {
     if (target.isFilled) {
+      clearedTargets += 1;
       target.clear(board, emptyValue);
+      var i = 0;
+      while (i < coreState.pieceStage.nextPieces.length &&
+        coreState.pieceStage.nextPieces[i].mainCell.type != CELL_TYPE.NORMAL) {
+        i += 1;
+      }
+      if (i < coreState.pieceStage.nextPieces.length) {
+        coreState.pieceStage.nextPieces[i].mainCell = target.mainCell;
+      }
     } else if (target.isGameOver) {
```

Comment:
else: target has been cleared but not filled, meaning that it has expired. Add a strike